cpu/stm32/stmclk/stmclk_l0l1.c

```diff
@@ -94,22 +193,57 @@ void stmclk_init_sysclk(void)
     PWR->CR = PWR_CR_VOS_0;
     /* Wait Until the Voltage Regulator is ready */
     while((PWR->CSR & PWR_CSR_VOSF) != 0) {}
-    /* HCLK = SYSCLK */
-    RCC->CFGR |= (uint32_t)CLOCK_AHB_DIV;
-    /* PCLK2 = HCLK */
-    RCC->CFGR |= (uint32_t)CLOCK_APB2_DIV;
-    /* PCLK1 = HCLK */
-    RCC->CFGR |= (uint32_t)CLOCK_APB1_DIV;
-    /*  PLL configuration: PLLCLK = CLOCK_SOURCE / PLL_DIV * PLL_MUL */
-    RCC->CFGR &= ~((uint32_t)(RCC_CFGR_PLLSRC | RCC_CFGR_PLLDIV | RCC_CFGR_PLLMUL));
-    RCC->CFGR |= (uint32_t)(CLOCK_PLL_SOURCE | CLOCK_PLL_DIV | CLOCK_PLL_MUL);
-    /* Enable PLL */
-    RCC->CR |= RCC_CR_PLLON;
-    /* Wait till PLL is ready */
-    while ((RCC->CR & RCC_CR_PLLRDY) == 0) {}
-    /* Select PLL as system clock source */
-    RCC->CFGR &= ~((uint32_t)(RCC_CFGR_SW));
-    RCC->CFGR |= (uint32_t)RCC_CFGR_SW_PLL;
-    /* Wait till PLL is used as system clock source */
-    while ((RCC->CFGR & (uint32_t)RCC_CFGR_SWS) != RCC_CFGR_SWS_PLL) {}
+
+    /* Enable HSE if needed */
+    if (IS_ACTIVE(CLOCK_ENABLE_HSE)) {
+        RCC->CR |= (RCC_CR_HSEON);
+        while (!(RCC->CR & RCC_CR_HSERDY)) {}
+    }
+
+    /* Enable MSI if needed */
+    if (IS_ACTIVE(CLOCK_ENABLE_MSI)) {
+        /* Configure MSI range and enable it */
+        RCC->ICSCR |= CLOCK_MSIRANGE;
+        RCC->CR |= (RCC_CR_MSION);
+        while (!(RCC->CR & RCC_CR_MSIRDY)) {}
+    }
+
+    /* Enable PLL if needed */
+    if (IS_ACTIVE(CLOCK_ENABLE_PLL)) {
+        /* Configure PLL clock source and configure the different prescalers */
+        RCC->CFGR &= ~(RCC_CFGR_PLLSRC | RCC_CFGR_PLLDIV | RCC_CFGR_PLLMUL);
+        RCC->CFGR |= (CLOCK_PLL_SOURCE | CLOCK_PLL_DIV | CLOCK_PLL_MUL);
+        /* Enable PLL */
+        RCC->CR |= RCC_CR_PLLON;
+        /* Wait till PLL is ready */
+        while (!(RCC->CR & RCC_CR_PLLRDY)) {}
+    }
+
+    /* Configure SYSCLK */
+    if (!IS_ACTIVE(CLOCK_ENABLE_HSI)) {
+        RCC->CFGR &= ~(RCC_CFGR_SW);
+    }
```

Comment:
This does not seem accurate as a comment, isn't it reseting the configuration?

Reply:
The comment refers to this if block and the if/else if below. All this logic is about selecting the right clock source for SYSCLK and clearing HSI if it's not used as SYSCLK (since it's already selected).